lang/machine.md

```diff
@@ -137,17 +123,44 @@ We also define some helper functions that will be useful later.
 ```rust
 impl<M: Memory> Machine<M> {
     fn cur_frame(&self) -> StackFrame<M> {
-        self.stack.last().unwrap()
+        let Some(active_thread) = self.thread_manager.active_thread else {
+            panic!("`cur_frame` called without active thread!");
+        };
+
+        self.thread_manager.threads[active_thread].cur_frame()
     }
 
     fn mutate_cur_frame<O>(&mut self, f: impl FnOnce(&mut StackFrame<M>) -> O) -> O {
-        if self.stack.is_empty() {
-            panic!("`mutate_cur_frame` called on empty stack!");
-        }
+        let Some(active_thread) = self.thread_manager.active_thread else {
+            panic!("`mutate_cur_frame` called without active thread!");
+        };
 
-        let last_idx = self.stack.len() - 1;
-        self.stack.mutate_at(last_idx, f)
+        self.thread_manager.threads.mutate_at(active_thread, |thread| thread.mutate_cur_frame(f))
     }
+
+    fn mutate_cur_stack<O>(&mut self, f: impl FnOnce(&mut List<StackFrame<M>>) -> O) -> O {
+        let Some(active_thread) = self.thread_manager.active_thread else {
+            panic!("`cur_stack` called without active thread!");
+        };
+
+        self.thread_manager.threads.mutate_at(active_thread, |thread| thread.mutate_stack(f))
+    }
+
+    fn cur_thread(&self) -> Thread<M> {
+        let Some(active_thread) = self.thread_manager.active_thread else {
+            panic!("`cur_thread` called without active thread!");
+        };
+
+        self.thread_manager.threads[active_thread]
+    }
+
+    fn mut_cur_thread<O>(&mut self, f: impl FnOnce(&mut Thread<M>) -> O) -> O {
```

Comment:
Should be called `mutate_cur_thread` for consistency with the other similar functions